config/parameters.js

```diff
@@ -12,17 +12,19 @@ const SCHEMA_PARAMETERS_OBJECT = Joi.object({
     'object.unknown': 'only supports string or key: {{#label}} pair as values'
 });
 
-const SCHEMA_PARAMETERS = Joi.object()
-    .pattern(
-        Joi.any(),
-        Joi.alternatives().try(
-            SCHEMA_PARAMETERS_STRING,
-            Joi.array().items(SCHEMA_PARAMETERS_STRING),
-            SCHEMA_PARAMETERS_OBJECT
-        )
+const SCHEMA_PARAMETERS_BASE = Joi.object().pattern(
+    Joi.any(),
+    Joi.alternatives().try(
+        SCHEMA_PARAMETERS_STRING,
+        Joi.array().items(SCHEMA_PARAMETERS_STRING),
+        SCHEMA_PARAMETERS_OBJECT
     )
-    .default({});
+);
```

Comment:
Do we need multiple schemas for default and optional?

Reply:
not needed, we should be having only one parameters schema

Reply:
Per discussion with @jithine, pipeline level parameters need not be changed (i.e. it should be not be made as optional).

In the parameters schema, we will export the base schema without `.default({})` or `.optional()`. In pipeline schema, we could extend the base schema to specify the default value `Parameters.parameters.default({})` and for jobs we could extend it to make it optional `Parameters.parameters.default({})`